core/src/main/scala/io/chrisdavenport/scalaz/task/instances/TaskInstances.scala

```diff
@@ -37,27 +50,25 @@ trait TaskInstances {
      * within the outer IO, discarding any error that might
      * occur
       **/
-    def runAsync[A](fa: Task[A])(cb: Either[Throwable, A] => IO[Unit]): IO[Unit] =
-      IO(
-        fa.unsafePerformAsync { disjunction =>
-          cb(disjunction.toEither)
-            .unsafeRunAsync(_ => ())
-        }
-      )
+    def runAsync[A](fa: Task[A])(cb: Either[Throwable, A] => IO[Unit]): SyncIO[Unit] =
+      SyncIO(cb(fa.unsafePerformSyncAttempt.toEither).unsafeRunAsyncAndForget())
```

Comment:
I'm not sure if this implementation makes any sense but it does make the tests pass.

Reply:
No, this implementation is not right, it shouldn't be blocking. But I can't seem to implement this method in any way that doesn't spuriously fail.